pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -903,30 +903,46 @@ template <class Precision> class StateVectorKokkos {
     }
 
     /**
-     * @brief Apply a PauliX operator to the state vector using a matrix
+     * @brief Templated method that applies special n-qubit gates.
      *
+     * @tparam functor_t Gate functor class for Kokkos dispatcher.
+     * @tparam nqubits Number of qubits.
      * @param wires Wires to apply gate to.
      * @param inverse Indicates whether to use adjoint of gate.
-     * @param params Parameters for this gate
+     * @param params parameters for this gate
      */
-    void
-    applyPauliX(const std::vector<size_t> &wires, bool inverse = false,
-                [[maybe_unused]] const std::vector<Precision> &params = {}) {
+    template <template <class, bool> class functor_t, int nqubits>
+    void applyGateFunctor(
+        const std::vector<size_t> &wires, bool inverse = false,
+        [[maybe_unused]] const std::vector<Precision> &params = {}) {
         auto &&num_qubits = getNumQubits();
-        assert(wires.size() == 1);
+        assert(wires.size() == nqubits);
```

Comment:
Could we replace with `PL_ASSERT` or equivalent?

Reply:
Yes. Should I replace all `assert` in the source code while I'm at it?